x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/create.ts

```diff
@@ -54,7 +53,17 @@ export const registerSiemRuleMigrationsCreateRoute = (
             const ctx = await context.resolve(['securitySolution']);
             const ruleMigrationsClient = ctx.securitySolution.getSiemRuleMigrationsClient();
 
-            await siemMigrationAuditLogger.logCreateMigration({ migrationId });
+            let migrationId: string;
+
+            if (!providedMigrationId) {
+              /** if new migration */
+              migrationId = await ruleMigrationsClient.data.migrations.create();
+              await siemMigrationAuditLogger.logCreateMigration({ migrationId });
```

Comment:
For "write" actions, we need to call the audit logger before taking action. For "read" actions, it is the opposite, the audit logger needs to be called after the action.

Reply:
I don't think it is so important to have the ID of the migration created in the audit logs. In this scenario, the migration ID is not passed in the parameters, it's generated internally by the API, so I think knowing that the user X requested a migration creation is enough.

When rules are uploaded to a migration is a different scenario, the migration ID is passed by the user, so it makes sense to have it in the audit logs.

Reply:
>so I think knowing that the user X requested a migration creation is enough.

We already know that by looking at the `endpoint` call and the `request params`. Audit log without `migrationId` does not add anything. Main purpose of logs is help troubleshoot any error at a later stage. This log will at least show the `migrationId` was created without any issues and laters errors can be linked to this particular `migrationId`. This is specifically helpful during concurrent `create` requests.

Additionally, If server is creating an artifact, it should log it somewhere. No?

Reply:
Then, for the same reason, we should also add the rule IDs to the audit logger when we create the rule migration documents, just in case it receives an update by ID later on, isn't it?

Honestly, I don't think we should go into that level of detail. The migration ID is created manually here and looks like it should be logged, but that's not the conventional way of doing it. If we were to follow this rule everywhere, we would always need to create all the document IDs manually, send them to the audit logger, and then create the documents passing the ID.

I am not so familiar with Audit Logger, but I don't think using it should condition so much the implementation of the route handler.

Reply:
@semd , thanks for the explanation. I have fixed it in https://github.com/elastic/kibana/pull/216164/commits/43b91c5fdbe23f1f1a6f98b935c9e6ccd12f928d